activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskRuntimeImpl.java

```diff
@@ -215,48 +209,80 @@ public Task release(ReleaseTaskPayload releaseTaskPayload) {
         }
+        
         org.activiti.engine.task.Task internalTask = getInternalTask(updateTaskPayload.getTaskId());
-
-        if (updateTaskPayload.getTaskName() != null) {
-            internalTask.setName(updateTaskPayload.getTaskName());
+           
+        if ((newValue=updateTaskPayload.getTaskName()) != null) {
+            oldValue=internalTask.getName();
+            if (!Objects.equals(oldValue,newValue)) {
+                updates++;
+                internalTask.setName(newValue);
+            }
         }
-        if (updateTaskPayload.getDescription() != null) {
-            internalTask.setDescription(updateTaskPayload.getDescription());
+        
+        if ((newValue=updateTaskPayload.getDescription()) != null) {
+            oldValue=internalTask.getDescription();
+            if (!Objects.equals(oldValue,newValue)) {
+                updates++;
+                internalTask.setDescription(newValue);
+            }
         }
+            
         if (updateTaskPayload.getPriority() != null) {
-            internalTask.setPriority(updateTaskPayload.getPriority());
-        }
-        if (updateTaskPayload.getAssignee() != null) {
-            internalTask.setAssignee(updateTaskPayload.getAssignee());
+            if (internalTask.getPriority()!=updateTaskPayload.getPriority()) {
+                updates++;
+                internalTask.setPriority(updateTaskPayload.getPriority());
+            }
         }
+        
         if (updateTaskPayload.getDueDate() != null) {
-            internalTask.setDueDate(updateTaskPayload.getDueDate());
+            if (!Objects.equals(internalTask.getDueDate(),updateTaskPayload.getDueDate())) {
+                updates++;
+                internalTask.setDueDate(updateTaskPayload.getDueDate());
+            }
         }
-        if (updateTaskPayload.getParentTaskId() != null) {
-            internalTask.setParentTaskId(updateTaskPayload.getParentTaskId());
+        
+        //@TODO: check if this value can be updated
+        if ((newValue=updateTaskPayload.getParentTaskId()) != null) {
+            oldValue=internalTask.getParentTaskId();
+            if (!Objects.equals(oldValue,newValue)) {
+                updates++;
+                internalTask.setParentTaskId(newValue);
+            }
         }
-        if (updateTaskPayload.getFormKey() != null) {
-            internalTask.setFormKey(updateTaskPayload.getFormKey());
+        
+        if ((newValue=updateTaskPayload.getFormKey()) != null) {
+            oldValue=internalTask.getFormKey();
+            if (!Objects.equals(oldValue,newValue)) {
+                updates++;
+                internalTask.setFormKey(newValue);
+            }
         }
-        //@TODO: add check to see if something was changed before saving + add all other updateable values
-        org.activiti.engine.task.Task updatedTask = taskService.saveTask(internalTask);
         
-        return taskConverter.from(updatedTask);
+        if (updates>0)
```

Comment:
@CTI777 same here.. coding style should use curly braces in all block statements